ocs_ci/utility/csr.py

```diff
@@ -169,3 +173,13 @@ def wait_for_all_nodes_csr_and_approve(timeout=900, sleep=10, expected_node_num=
         pending_csrs = get_pending_csr()
         if pending_csrs:
             approve_csrs(pending_csrs)
+        # In vSphere deployment it sometime happes that VM doesn't get ip and
+        # then we need to restart it to make our CI more stable and let the VM
+        # to get IP and continue with loading ignition config. The rester of
+        # the VMs happens only once in reboot_timeout (120 seconds).
+        if (
+            config.ENV_DATA["platform"] == constants.VSPHERE_PLATFORM
+            and start_time >= reboot_timeout
+        ):
+            start_time = time.time()
+            found_vms_without_ip_and_restart()
```

Comment:
we need to give time for CSR's to genearte. placing this function here doesn't give time for CSRs to generate and we are rebooting the node

Reply:
120 is enough